src/storage/src/source/kafka.rs

```diff
@@ -709,17 +703,20 @@ impl PartitionConsumer {
     fn get_next_message(
         &mut self,
     ) -> Result<
-        Result<Option<SourceMessage<Option<Vec<u8>>, Option<Vec<u8>>, ()>>, KafkaError>,
-        anyhow::Error,
+        Option<(
+            SourceMessage<Option<Vec<u8>>, Option<Vec<u8>>>,
+            (PartitionId, MzOffset),
+        )>,
+        KafkaError,
```

Comment:
we ultimately panic on this right?

Reply:
This `get_next_message` is called from `poll_from_next_queue` above, which pats this to a `String`, which gets passed up to the `SourceReader::get_next_message`, which reports it as a source status. So this won't cause a panic, but it _will_ get reported to the user in some way.